task5/src/main/java/com/clouway/adapter/persistence/PersistentSessionRepository.java

```diff
@@ -0,0 +1,153 @@
+      statement.execute();
+    } catch (SQLException sqlExc) {
+      sqlExc.printStackTrace();
+    } finally {
+      try {
+        statement.close();
+      } catch (SQLException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  public Session get(String sessionID) {
+    Session session = null;
+    Connection connection = connectionProvider.get();
+    PreparedStatement statement = null;
+    try {
+      statement = connection.prepareStatement("SELECT * FROM session WHERE sessionID=?");
+      statement.setString(1, sessionID);
+      ResultSet resultSet = statement.executeQuery();
+      while (resultSet.next()) {
+        session = new Session(resultSet.getString("email"), resultSet.getString("sessionID"));
+      }
+      resultSet.close();
+    } catch (SQLException e) {
+      e.printStackTrace();
+    } finally {
+      try {
+        statement.close();
+      } catch (SQLException e) {
+        e.printStackTrace();
+      }
+    }
+    return session;
+  }
+
+  public void delete(String sessionID) {
+    Connection connection = connectionProvider.get();
+    PreparedStatement statement = null;
+    try {
+      statement = connection.prepareStatement("DELETE FROM session WHERE sessionID=?");
+      statement.setString(1, sessionID);
+      statement.execute();
+    } catch (SQLException e) {
+      e.printStackTrace();
+    } finally {
+      try {
+        statement.close();
+      } catch (SQLException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  public void deleteAll() {
+    Connection connection = connectionProvider.get();
+    PreparedStatement statement = null;
+    try {
+      statement = connection.prepareStatement("DELETE FROM session");
+      statement.execute();
+    } catch (SQLException e) {
+      e.printStackTrace();
+    } finally {
+      try {
+        statement.close();
+      } catch (SQLException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  public void cleanExpired() {
+    Connection connection = connectionProvider.get();
+    PreparedStatement statement = null;
+    try {
+      statement = connection.prepareStatement("DELETE FROM session WHERE expirationTime < " + System.currentTimeMillis());
+      statement.execute();
+    } catch (SQLException e) {
+      e.printStackTrace();
+    } finally {
+      try {
+        statement.close();
+      } catch (SQLException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  public int getOnlineUsersNumber() {
+    Connection connection = connectionProvider.get();
+    ResultSet resultSet = null;
+    PreparedStatement statement;
+    Integer counter = null;
+    try{
+      statement = connection.prepareStatement("SELECT COUNT(*) FROM session");
+      resultSet = statement.executeQuery();
+      while (resultSet.next()){
+        counter=resultSet.getInt(1);
+      }
+    } catch (SQLException e) {
+      e.printStackTrace();
+    }
+    return counter;
+  }
+
+  public void resetSessionTime(Session session){
```

Comment:
`refresh()` You refresh session with 5 more minutes. `reset` is not propper word in this case